### PR TITLE
Auth signup

### PR DESCRIPTION
**Pull request by saketkum11**
- signup
- require Auth
- added auth to videoList Page. 
- page "/videoList"

### CODE REVIEW

File: src/Context/Auth-context/Auth-context.js [hunk 1]
```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
...
     try {
-      const response = await axios.post(`/api/auth/login`, {
-        email: email,
-        password: password,
+      const response = await axios.post("/api/auth/signup", {
+        email: data.email,
+        password: data.password,
+        firstName: data.firstName,
+        lastName: data.lastName,
       });
```
Comment (Sharmaryan) [inline src/Context/Auth-context/Auth-context.js:20]: you can destructure these values from data itslef, ``` const response = await axios.post("/api/auth/signup", {
        email,
        password,
        firstName,
        lastName,
      }); ```
and then there will be no need of writing email:email .

File: src/Context/Auth-context/Auth-context.js [hunk 2]
```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
...
+      console.error(error);
+    }
+  };
+  const loginHandler = async ({ email, password }) => {
+    if (isAuth) {
+      try {
+        const response = await axios.post("/api/auth/login", {
+          email: email,
+          password: password,
+        });
```
Comment (Sharmaryan) [inline src/Context/Auth-context/Auth-context.js:33]: when key and value pair are same, then you can mention key only. as I mentioned in the above code

File: src/Context/Auth-context/Auth-context.js [hunk 3]
```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
...
+        // saving the encodedToken in the localStorage
+        localStorage.setItem("token", response.data.encodedToken);
+        console.log("response from login", response);
+      } catch (error) {
```
Comment (Sharmaryan) [inline src/Context/Auth-context/Auth-context.js:36]: now work is done you can remove console log

File: src/Context/Auth-context/Auth-context.js [hunk 4]
```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
...
+    } else {
+      console.log("signup");
     }
   };
 
+  console.log("usedata", credentialData);
   return (
-    <authContext.Provider value={{ loginHandler }}>
+    <authContext.Provider
+      value={{
```
Comment (Sharmaryan) [inline src/Context/Auth-context/Auth-context.js:47]: I don't know it is mandatory or not but  authContext should be written as AuthContext

**Review by Sharmaryan — APPROVED** [review]

File: src/Context/Auth-context/Auth-context.js (hunk 4 above)
Comment (saketkum11) [reply src/Context/Auth-context/Auth-context.js:47]: okk thank you for reminding.

File: src/Context/Auth-context/Auth-context.js (hunk 3 above)
Comment (saketkum11) [reply src/Context/Auth-context/Auth-context.js:36]: yes   sure

File: src/Context/Auth-context/Auth-context.js (hunk 2 above)
Comment (saketkum11) [reply src/Context/Auth-context/Auth-context.js:33]: okk this is somthing new for me.

File: src/Context/Auth-context/Auth-context.js (hunk 1 above)
Comment (saketkum11) [reply src/Context/Auth-context/Auth-context.js:20]: yes I have changed

**Review by JowelTisso — APPROVED** [review]
* Other than @Sharmaryan suggestions
* Rest looks good

**Pull request merged**